testsuite/forge/src/success_criteria.rs

```diff
@@ -29,52 +36,78 @@ pub struct SuccessCriteria {
 }
 
 impl SuccessCriteria {
-    pub fn new(
-        tps: usize,
-        max_latency_ms: usize,
-        check_no_restarts: bool,
-        wait_for_all_nodes_to_catchup: Option<Duration>,
-        system_metrics_threshold: Option<SystemMetricsThreshold>,
-        chain_progress_check: Option<StateProgressThreshold>,
-    ) -> Self {
+    pub fn new(tps: usize) -> Self {
         Self {
             avg_tps: tps,
-            max_latency_ms,
-            check_no_restarts,
-            wait_for_all_nodes_to_catchup,
-            system_metrics_threshold,
-            chain_progress_check,
+            latency_thresholds: Vec::new(),
+            check_no_restarts: false,
+            wait_for_all_nodes_to_catchup: None,
+            system_metrics_threshold: None,
+            chain_progress_check: None,
         }
     }
 
+    pub fn add_no_restarts(mut self) -> Self {
```

Comment:
nit: A builder pattern for success criteria might be cleaner intsead of passing `mut self` here.

Reply:
with Rust and ownership model, that is effectively the same thing, no?